From 5c55d79ecfecc4a1b5a5ebcf649227ca1e1bf38e Mon Sep 17 00:00:00 2001
From: Joao Lima <joao.lima@hbkworld.com>
Date: Tue, 30 Sep 2025 12:13:04 +0000
Subject: [PATCH 1/2] tcl/board/ti_*_swd_native.cfg: Add support for direct
 memory access via SW

Add support for SWD emulation as a transport method for
direct memory operations of boards TI AM64x and TI J784s4

Upstream-Status: Submitted [http://openocd.zylin.com/#/c/9020/]

Change-Id: I17fe9b2bef5c58886625bfdb88d92645ba4d7da7
Signed-off-by: Joao Lima <joao.lima@hbkworld.com>
---
 tcl/board/ti_am64xx_swd_native.cfg | 23 +++++++++++++++++++++++
 tcl/board/ti_j784s4_swd_native.cfg | 22 ++++++++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 tcl/board/ti_am64xx_swd_native.cfg
 create mode 100644 tcl/board/ti_j784s4_swd_native.cfg

diff --git a/tcl/board/ti_am64xx_swd_native.cfg b/tcl/board/ti_am64xx_swd_native.cfg
new file mode 100644
index 000000000..d3727149e
--- /dev/null
+++ b/tcl/board/ti_am64xx_swd_native.cfg
@@ -0,0 +1,23 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2022-2023 Texas Instruments Incorporated - http://www.ti.com/
+#
+# Texas Instruments am642
+# Link: https://www.ti.com/product/AM642
+#
+# This configuration file is used as a self hosted debug configuration that
+# works on every AM642 platform based on firewall configuration permitted
+# in the system.
+#
+# In this system openOCD runs on one of the CPUs inside AM625 and provides
+# network ports that can then be used to debug the microcontrollers on the
+# SoC - either self hosted IDE OR remotely.
+
+# We are using dmem, which uses dapdirect_swd transport
+adapter driver dmem
+transport select swd
+
+if { ![info exists SOC] } {
+	set SOC am642
+}
+
+source [find target/ti_k3.cfg]
diff --git a/tcl/board/ti_j784s4_swd_native.cfg b/tcl/board/ti_j784s4_swd_native.cfg
new file mode 100644
index 000000000..13b2ac3b8
--- /dev/null
+++ b/tcl/board/ti_j784s4_swd_native.cfg
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (C) 2022-2023 Texas Instruments Incorporated - http://www.ti.com/
+#
+# Texas Instruments TDA4VM/J721E
+# Link: https://www.ti.com/product/TDA4VM
+#
+# This configuration file is used as a self hosted debug configuration that
+# works on every TDA4VM platform based on firewall configuration permitted
+# in the system.
+#
+# In this system openOCD runs on one of the CPUs inside TDA4VM and provides
+# network ports that can then be used to debug the microcontrollers on the
+# SoC - either self hosted IDE OR remotely.
+
+# We are using dmem, which uses dapdirect_swd transport
+adapter driver dmem
+transport select swd
+
+if { ![info exists SOC] } {
+        set SOC j784s4
+}
+source [find target/ti_k3.cfg]
